webrender/src/picture.rs

```diff
@@ -897,6 +915,20 @@ impl TileCache {
 
         // Step through each tile and invalidate if the dependencies have changed.
         for (i, tile) in self.tiles.iter_mut().enumerate() {
+            // Invalidate if the backing texture was evicted.
+            if !resource_cache.texture_cache.is_allocated(&tile.handle) {
+                tile.is_valid = false;
+            }
+
+            // Request the backing texture so it won't get evicted this frame.
+            // We specifically want to mark the tile texture as used, even
+            // if it's detected not visible below and skipped. This is because
+            // we maintain the set of tiles we care about based on visibility
+            // during pre_update. If a tile still exists after that, we are
+            // assuming that it's either visible or we want to retain it for
+            // a while in case it gets scrolled back onto screen soon.
+            resource_cache.texture_cache.request(&tile.handle, gpu_cache);
```

Comment:
I think it would be clearer if this were in an else block, since it's a no-op in the already-evicted case (handled above), but that's not obvious to readers.

Reply:
Also, it seems like what we really want here is EvictionPolicy::Manual, which would let us skip this work, and trigger faster eviction of the tiles we don't care about.

I won't insist you do this now, but add a TODO?